src/USER-DPD/fix_shardlow.cpp

```diff
@@ -155,12 +156,13 @@ void FixShardlow::setup(int /*vflag*/)
   bool fixShardlow = false;
 
   for (int i = 0; i < modify->nfix; i++)
-    if (strncmp(modify->fix[i]->style,"nvt",3) == 0 || strncmp(modify->fix[i]->style,"npt",3) == 0)
-      error->all(FLERR,"Cannot use constant temperature integration routines with DPD.");
+    if (strstr(modify->fix[i]->style,"nvt") || strstr(modify->fix[i]->style,"npt") ||
+        strstr(modify->fix[i]->style,"gle") || strstr(modify->fix[i]->style,"gld"))
```

Comment:
Shouldn't those start with `^` to match it as a prefix?

Reply:
> Shouldn't those start with `^` to match it as a prefix?

No, those use `strstr()` because in this case the check was not general enough. This restriction also applies to cases where the string is elsewhere in the style name, e.g. for "rigid/nvt" or "rigid/nvt/omp". This is risking a false positive, but this is a rather specialized feature of LAMMPS and not likely to be used with features where such a conflict is likely.